### DISCUSSION TITLE
Testing Golang Manual Tracing skywalking-go v0.3.0

### DISCUSSION THREAD

**Discussion by Ecostack** [apache/skywalking#11447] [General]
Been testing the new version and I would like to say, thanks for this addition. It works really well on a first test. I would like to report a couple of things though, please let me know if this is the right place to post or not.

### Toolkit/Trace API
The package github.com/apache/skywalking-go v0.3.0 does not expose the toolkit/trace API, it is a separate package.

Would it not be better if that is both coming from the skywalking-go package?

```
require (
 	github.com/apache/skywalking-go v0.3.0
	github.com/apache/skywalking-go/toolkit v0.0.0-20231019114202-ff7ad0ddecba
)
```

### Injector & Extractor Ref
There are two methods, `CreateEntrySpan(operationName string, extractor ExtractorRef)` and `CreateExitSpan(operationName string, peer string, injector InjectorRef)` which require either an **injector or extractor ref**. 
How would I obtain these or use these methods? Maybe provide an example of how these can be used.

### Component unknown
Manual span entries have component value set to `unknown`. Can this be set manually or would it be possible to set this value somehow automatic, maybe the file name or such?

**Comment by wu-sheng** [comment]
@Alipebt @mrproliu Let's continue the discussion here.

**Comment by mrproliu** [comment]
>The package github.com/apache/skywalking-go v0.3.0 does not expose the toolkit/trace API, it is a separate package. 

I have added a new tag for getting the `toolkit` package. 

> How would I obtain these or use these methods? Maybe provide an example of how these can be used.

We already [have a demo in there](https://github.com/apache/skywalking-go/blob/main/test/plugins/scenarios/trace-activation/test_service.go#L80-L87). @Alipebt Can you update this section of the document? Make it more easy to understand. 

> Manual span entries have component value set to unknown. Can this be set manually or would it be possible to set this value somehow automatic, maybe the file name or such?

The extension of the component may be relatively complex, and you need to save the component ID in the backend. You can put the relevant description of the component into the name of Span.

**Reply by Alipebt** [reply]
Ok, I will do it

**Reply by wu-sheng** [reply]
> The extension of the component may be relatively complex, and you need to save the component ID in the backend. You can put the relevant description of the component into the name of Span.

I think the component ID could be bypassed through toolkit once the end users feel it is safe to use.
Ref protocol, https://github.com/apache/skywalking-data-collect-protocol/blob/d4da5699915ee52288f8ff1c954decf6363485bc/language-agent/Tracing.proto#L147-L153

Pre-defined component IDs are here, https://skywalking.apache.org/docs/main/next/en/guides/component-library-settings/

If users pick one, and update the UI, it works. But most likely, they may want to pick an existing one, and UI could work out-of-the-box.

**Reply by mrproliu** [reply]
> I think the component ID could be bypassed through toolkit once the end users feel it is safe to use.

Sure, @Alipebt could you help to add this API?

**Reply by Alipebt** [reply]
Sure，I'll update that with the documentation
